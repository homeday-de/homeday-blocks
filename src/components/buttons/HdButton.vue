<template>
  <button :class="computedClasses">
    <slot />
  </button>
</template>

<script>
export const TYPES = {
  DEFAULT: '',
  PRIMARY: 'primary',
  SECONDARY: 'secondary',
  TERTIARY: 'tertiary',
  FLAT: 'flat',
  GHOST: 'ghost',
};

export default {
  name: 'HdButton',
  props: {
    modifier: {
      type: String,
      default: TYPES.DEFAULT,
      validator(value) {
        const allTypes = Object.values(TYPES);

        return allTypes.includes(value);
      },
    },
  },
  computed: {
    computedClasses() {
      const baseClass = 'btn';
      const classes = [baseClass];

      if (this.modifier) {
        classes.push(`${baseClass}--${this.modifier}`);
      }

      return classes;
    },
  },
};
</script>

<style lang="scss" scoped>
.btn {
  width: 100%;
}
</style>
