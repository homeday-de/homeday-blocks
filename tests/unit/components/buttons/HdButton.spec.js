import { shallowMount } from '@vue/test-utils';
import HdButton, { TYPES } from '@/components/buttons/HdButton.vue';

describe('HdButton', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(HdButton, {
      slots: {
        default: '<span>Button text</span>',
      },
    });
  });

  it('should render component with `btn` class', () => {
    expect(wrapper.classes()).toContain('btn');
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should render component with `btn--primary` class', () => {
    const modifier = TYPES.PRIMARY;
    const className = `btn--${modifier}`;

    wrapper.setProps({
      modifier,
    });

    expect(wrapper.classes()).toContain(className);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should render component with `btn--secondary` class', () => {
    const modifier = TYPES.SECONDARY;
    const className = `btn--${modifier}`;

    wrapper.setProps({
      modifier,
    });

    expect(wrapper.classes()).toContain(className);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should render component with `btn--tertiary` class', () => {
    const modifier = TYPES.TERTIARY;
    const className = `btn--${modifier}`;

    wrapper.setProps({
      modifier,
    });

    expect(wrapper.classes()).toContain(className);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should render component with `btn--flat` class', () => {
    const modifier = TYPES.FLAT;
    const className = `btn--${modifier}`;

    wrapper.setProps({
      modifier,
    });

    expect(wrapper.classes()).toContain(className);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should render component with `btn--ghost` class', () => {
    const modifier = TYPES.GHOST;
    const className = `btn--${modifier}`;

    wrapper.setProps({
      modifier,
    });

    expect(wrapper.classes()).toContain(className);
    expect(wrapper.html()).toMatchSnapshot();
  });
});
