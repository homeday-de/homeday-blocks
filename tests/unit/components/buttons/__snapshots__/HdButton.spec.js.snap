// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HdButton should render component with \`btn\` class 1`] = `<button class="btn"><span>Button text</span></button>`;

exports[`HdButton should render component with \`btn--flat\` class 1`] = `<button class="btn btn--flat"><span>Button text</span></button>`;

exports[`HdButton should render component with \`btn--ghost\` class 1`] = `<button class="btn btn--ghost"><span>Button text</span></button>`;

exports[`HdButton should render component with \`btn--primary\` class 1`] = `<button class="btn btn--primary"><span>Button text</span></button>`;

exports[`HdButton should render component with \`btn--secondary\` class 1`] = `<button class="btn btn--secondary"><span>Button text</span></button>`;

exports[`HdButton should render component with \`btn--tertiary\` class 1`] = `<button class="btn btn--tertiary"><span>Button text</span></button>`;
